### PR TITLE
Separate annotations from labels in discovered pods

### PR DESCRIPTION
**Pull request by jrcamp**
Before annotations were merged into labels which could be confusing. Separate
them out. Also document how to use these annotations in receiver creator.

### CODE REVIEW

File: receiver/receivercreator/README.md [hunk 1]
```diff
@@ -67,18 +69,25 @@ extensions:
   k8s_observer:
 
 receivers:
+  prometheus_simple:
+    # Configure prometheus scraping if standard prometheus annotations are set on the pod.
+    rule: type.pod && annotations["prometheus.io/scrape"] == "true"
+    config:
+      metrics_path: '`"prometheus.io/path" in annotations ? annotations["prometheus.io/path"] : "/metrics"`'
+      endpoint: '`endpoint`:`"prometheus.io/port" in annotations ? annotations["prometheus.io/port"] : 9090`'
```
Comment (dmitryax) [inline receiver/receivercreator/README.md:77]: Looks like it's put under a wrong `receivers` section. should be under `receiver_creator/1` -> `receivers`. right?

Comment (jrcamp) [reply receiver/receivercreator/README.md]: Yep, thanks

**Review by dmitryax — APPROVED** [review]
Nice!

**Comment by jrcamp** [comment]
@tigrannajaryan please merge

**Pull request merged**